src/main/java/com/example/auctionapp/service/SubcategoryService.java

```diff
@@ -0,0 +1,114 @@
+package com.example.auctionapp.service;
+
+import com.example.auctionapp.dto.SubcategoryDto;
+import com.example.auctionapp.exception.NotFoundException;
+import com.example.auctionapp.model.Category;
+import com.example.auctionapp.model.Subcategory;
+import com.example.auctionapp.repository.BaseRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@Transactional
+public class SubcategoryService implements IBaseService<SubcategoryDto> {
+
+    BaseRepository<Subcategory> repository;
+    BaseRepository<Category> categoryRepository;
+
+    private Logger logger = LoggerFactory.getLogger(SubcategoryService.class);
+
+    @Autowired
+    public SubcategoryService(BaseRepository<Subcategory> repository,
+                              BaseRepository<Category> categoryRepository) {
+        this.repository = repository;
+        this.repository.setResourceClass(Subcategory.class);
+        this.categoryRepository=categoryRepository;
+        this.categoryRepository.setResourceClass(Category.class);
+    }
+
+
+    public List<SubcategoryDto> getAll() {
+
+        List<Subcategory> subcategories = repository.findAll();
+        List<SubcategoryDto> subcategoryDtos = new ArrayList<>();
+
+        for (Subcategory subcategory:subcategories) {
+            subcategoryDtos.add(new SubcategoryDto(subcategory.getId(), subcategory.getDateCreated(), subcategory.getLastModifiedDate(), subcategory.getName(), subcategory.getCategory().getId()));
+        }
+
+        return  subcategoryDtos;
+    }
+
+
+    public SubcategoryDto getById(Long id) {
+
+        Subcategory subcategory = repository.findById(id);
+        if(subcategory == null) {
+            String message = "Subcategory with id " + id + " does not exist";
+            logger.error(message);
+            throw new NotFoundException(message);
+        }
```

Comment:
I see that this specific part of code is being repeated from method to method in every type of resource service.
One rule in programming is that if a specific logic is repeated, it can and should be extracted somewhere.
You can create a utility class that has a generic method to which you send a repository and an id, and it sends returns back the requested object or throws a method.